src/Saas.Admin/Saas.Admin.Service/Data/TenantDbInitializer.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Saas.Admin.Service.Data;
+
+public static class TenantDbInitializer
+{
+
+    public static void ConfigureDatabase(this IHost host)
+    {
+        using IServiceScope scope = host.Services.CreateScope();
+        
+        ILogger logger = scope.ServiceProvider.GetRequiredService<ILogger<TenantsContext>>();
+        TenantsContext tenantsContext = scope.ServiceProvider.GetRequiredService<TenantsContext>();
+
+        CreateDatabase(tenantsContext, logger);
+        SeedDatabase(tenantsContext, logger);
+    }
+
+    private static void CreateDatabase(TenantsContext tenantsContext, ILogger logger)
+    {
+        try
+        {
+            tenantsContext.Database.EnsureCreated();
+        }
+        catch (Exception ex)
+        {
+            logger.LogCritical(ex, "Unable to create the database");
+            throw;
+        }
+    }
+
+    private static void SeedDatabase(TenantsContext tenantsContext, ILogger logger)
+    {
+        try
+        {
+            //Add any code required to seed the database here
```

Comment:
Do we need to add code here to ensure that it's only seeded if the database was just created? IE we don't want to seed twice.

Reply:
Good point, I'll add a check here